src/instruments/src/ATC/index.tsx

```diff
@@ -0,0 +1,110 @@
+import './style.scss';
+import React, { useEffect, useState } from 'react';
+import { render } from '../Common';
+import { useSimVar, useSplitSimVar } from '../Common/simVars';
+import { useInteractionEvent, useUpdate } from '../Common/hooks';
+
+const getDigitsFromCode = (code: number): number[] => {
+    let codeCopy = code;
+    const digits: number[] = [];
+    while (codeCopy > 0) {
+        digits.push(codeCopy % 10);
+        codeCopy = Math.floor(codeCopy / 10);
+    }
+    if (digits.length < 4) {
+        for (let i = 0; i <= 4 - digits.length; i++) {
+            digits.push(0);
+        }
+    }
+    digits.reverse();
+    return digits;
+};
+const getBco16FromDigits = (digits: number[]): number => (digits[0] * 4096) + (digits[1] * 256) + (digits[2] * 16) + digits[3];
+
+const PoweredXpdrDisplay = () => {
+    const [newDigits, setNewDigits] = useState<null | number[]>(null);
+    const [doubleClrTimer, setDoubleClrTimer] = useState(-1);
+    const [displayResetTimer, setDisplayResetTimer] = useState(-1);
+    const [ltsTest] = useSimVar('L:XMLVAR_LTS_Test', 'Bool', 250);
+
+    const [transponderCode, setTransponderCode] = useSplitSimVar('TRANSPONDER CODE:1', 'number', 'K:XPNDR_SET', 'Bco16');
+    const codeInDisplay = newDigits !== null ? newDigits : getDigitsFromCode(transponderCode);
+
+    useUpdate((deltaTime) => {
+        if (doubleClrTimer > 0) {
+            setDoubleClrTimer(Math.max(doubleClrTimer - deltaTime / 1000, 0));
+        }
+        if (displayResetTimer > 0) {
+            setDisplayResetTimer(Math.max(displayResetTimer - deltaTime / 1000, 0));
+        }
+    });
```

Comment:
Might be worth having a discussion on using setTimeout for something like this. It's probably the perfect example where the code would be much cleaner and it would probably work just as well (https://coherent-labs.com/Documentation/cpp-gt/d9/d28/time_control.html).